introspection-engine/introspection-engine-tests/src/lib.rs

```diff
@@ -1,51 +1,56 @@
+pub mod test_api;
+
 use barrel::Migration;
-use pretty_assertions::assert_eq;
-use quaint::{connector::Queryable, single::Quaint};
+use quaint::{prelude::Queryable, single::Quaint};
 
-pub(crate) fn custom_assert(left: &str, right: &str) {
-    let parsed_expected = datamodel::parse_datamodel(&right).unwrap();
-    let reformatted_expected =
-        datamodel::render_datamodel_to_string(&parsed_expected).expect("Datamodel rendering failed");
+#[macro_export]
+macro_rules! assert_eq_datamodels {
+    ($left:expr, $right:expr) => {{
+        let parsed_result = datamodel::parse_datamodel($left).unwrap();
+        let parsed_expected = datamodel::parse_datamodel($right).unwrap();
 
-    assert_eq!(left, reformatted_expected);
+        let reformatted_result = datamodel::render_datamodel_to_string(&parsed_result).unwrap();
+        let reformatted_expected = datamodel::render_datamodel_to_string(&parsed_expected).unwrap();
+
+        pretty_assertions::assert_eq!(reformatted_result, reformatted_expected);
+    }};
 }
 
-pub(crate) fn custom_assert_with_config(left: &str, right: &str) {
+pub fn custom_assert_with_config(left: &str, right: &str) {
     let parsed_expected_datamodel = datamodel::parse_datamodel(&right).unwrap();
     let parsed_expected_config = datamodel::parse_configuration(&right).unwrap();
+
     let reformatted_expected =
         datamodel::render_datamodel_and_config_to_string(&parsed_expected_datamodel, &parsed_expected_config)
             .expect("Datamodel rendering failed");
 
     assert_eq!(left, reformatted_expected);
 }
 
-pub(crate) fn assert_eq_json(a: &str, b: &str) {
+pub fn assert_eq_json(a: &str, b: &str) {
     let json_a: serde_json::Value = serde_json::from_str(a).expect("The String a was not valid JSON.");
     let json_b: serde_json::Value = serde_json::from_str(b).expect("The String b was not valid JSON.");
 
     assert_eq!(json_a, json_b);
 }
 
-// barrel
-
 pub struct BarrelMigrationExecutor {
-    pub(super) database: Quaint,
-    pub(super) sql_variant: barrel::backend::SqlVariant,
-    pub(super) schema_name: String,
+    database: Quaint,
+    sql_variant: barrel::backend::SqlVariant,
+    schema_name: String,
 }
 
 impl BarrelMigrationExecutor {
     pub async fn execute<F>(&self, migration_fn: F)
     where
-        F: FnOnce(&mut Migration),
+        F: FnOnce(&mut Migration) ,
```

Comment:
nitpick: these commas are a bit weird (the space)

Reply:
Coming from clippy btw... I should remove those.